### PR TITLE
FIX: class for section link when name has space

### PR DESCRIPTION
**Pull request by lis2**
Sidebar section link has class name based on link title. Title can contain spaces, therefore they should be replaced.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/sidebar/section-link.js [hunk 1]
```diff
@@ -28,7 +28,11 @@ export default class SectionLink extends Component {
     let classNames = ["sidebar-section-link", "sidebar-row"];
 
     if (this.args.linkName) {
-      classNames.push(`sidebar-section-link-${this.args.linkName}`);
+      classNames.push(
+        `sidebar-section-link-${this.args.linkName
+          .replace(" ", "-")
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/app/components/sidebar/section-link.js:33]: You actually need to use `replaceAll`, this just does one ;)

File: app/assets/javascripts/discourse/tests/integration/components/sidebar/section-link-test.js [hunk 2]
```diff
@@ -29,14 +29,14 @@ module("Integration | Component | sidebar | section-link", function (hooks) {
...
-        "sidebar-section-link-test",
+        "sidebar-section-link-test-meta",
       ],
       "has the right class attribute for the link"
     );
   });
 
   test("custom class attribute for link", async function (assert) {
-    const template = hbs`<Sidebar::SectionLink @linkName="test" @route="discovery.latest" @class="123 abc" />`;
+    const template = hbs`<Sidebar::SectionLink @linkName="Test Meta" @route="discovery.latest" @class="123 abc" />`;
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/tests/integration/components/sidebar/section-link-test.js:39]: Based on my other comment you need a test with 3 spaces for link name :)

**Review by martin-brennan — APPROVED** [review]

**Comment by martin-brennan** [comment]
@lis2 just need to fix up comments before merge but I approved

File: app/assets/javascripts/discourse/app/components/sidebar/section-link.js (hunk 1 above)
Comment (lis2) [reply app/assets/javascripts/discourse/app/components/sidebar/section-link.js]: great catch! Thank you :)

**Pull request merged**